tasks.c

```diff
@@ -2854,15 +2854,18 @@ void vTaskStartScheduler( void )
          * starts to run. */
         portDISABLE_INTERRUPTS();
 
-        #if ( configUSE_NEWLIB_REENTRANT == 1 )
+        #if ( ( configUSE_NEWLIB_REENTRANT == 1 ) && ( configNEWLIB_REENTRANT_IS_DYNAMIC == 0 ) )
```

Comment:
Can we avoid adding a new config option? Is it right to say that using newlib without __DYNAMIC_REENT__ with SMP is a bug? If so, we can just use configNUM_CORES instead?

```
#if ( ( configUSE_NEWLIB_REENTRANT == 1 ) && ( configNUM_CORES == 1 ) )
```

Reply:
@aggarg That work too. I'll update the check and comments.

Reply:
That is more confusing - let us keep `configNEWLIB_REENTRANT_IS_DYNAMIC` then.